quickwit-actors/src/actor.rs

```diff
@@ -0,0 +1,287 @@
+    pub fn record_progress(&self) {
+        self.progress.record_progress();
+    }
+
+    pub(crate) fn get_state(&self) -> ActorState {
+        self.actor_state.get_state()
+    }
+
+    pub(crate) fn pause(&mut self) {
+        self.actor_state.pause();
+    }
+
+    pub(crate) fn resume(&mut self) {
+        self.actor_state.resume();
+    }
+
+    pub(crate) fn terminate(&mut self, termination: &ActorTermination) {
+        if termination.is_failure() {
+            self.kill_switch().kill();
+        }
+        self.actor_state.terminate();
+    }
+}
+
+impl<A: SyncActor> ActorContext<A> {
+    /// Sends a message to the actor being the mailbox.
+    ///
+    /// This method hides logic to prevent an actor from being identified
+    /// as frozen if the destination actor channel is saturated, and we
+    /// are simply experiencing back pressure.
+    pub fn send_message_blocking<M: fmt::Debug>(
+        &self,
+        mailbox: &Mailbox<M>,
+        msg: M,
+    ) -> Result<(), crate::SendError> {
+        let _guard = self.protect_zone();
+        debug!(from=%self.self_mailbox.actor_instance_name(), to=%mailbox.actor_instance_name(), msg=?msg, "send");
+        mailbox.send_message_blocking(msg)
+    }
+
+    /// Sends a message to itself.
+    /// Since it is very easy to deadlock an actor, the behavior is quite
+    /// different from `send_message_blocking`.
+    ///
+    /// The method not
+    pub fn send_self_message_blocking(&self, msg: A::Message) -> Result<(), crate::SendError> {
+        debug!(self=%self.self_mailbox.actor_instance_name(), msg=?msg, "self_send");
+        self.self_mailbox
+            .sender
+            .try_send(ActorMessage::Message(msg))
+            .map_err(|_| crate::SendError::WouldDeadlock)?;
+        Ok(())
+    }
+
+    pub fn schedule_self_msg_blocking(&self, after_duration: Duration, msg: A::Message) {
+        let self_mailbox = self.inner.self_mailbox.clone();
+        let cmd_schedule_msg = Command::ScheduledMessage(msg);
+        let scheduler_msg = SchedulerMessage::ScheduleEvent {
+            timeout: after_duration,
+            callback: Callback(Box::pin(async move {
+                let _ = self_mailbox.send_command(cmd_schedule_msg).await;
+            })),
+        };
+        let _ = self.send_message_blocking(&self.inner.scheduler_mailbox, scheduler_msg);
+    }
+}
+
+impl<A: AsyncActor> ActorContext<A> {
+    /// `async` version of `send_message`
+    pub async fn send_message<M: fmt::Debug>(
+        &self,
+        mailbox: &Mailbox<M>,
+        msg: M,
+    ) -> Result<(), crate::SendError> {
+        let _guard = self.protect_zone();
+        debug!(from=%self.self_mailbox.actor_instance_name(), send=%mailbox.actor_instance_name(), msg=?msg);
+        mailbox.send_message(msg).await
+    }
+
+    /// `async` version of `send_self_message`
+    pub async fn send_self_message(&self, msg: A::Message) -> Result<(), crate::SendError> {
```

Comment:
send_self_command_message?

Reply:
I don't understand your comment.

Reply:
It looked like it was scheduling a command message